py/makeqstrdefs.py

```diff
@@ -39,11 +40,26 @@ def preprocess():
         os.makedirs(os.path.dirname(args.output[0]))
     except OSError:
         pass
-    with open(args.output[0], "w") as out_file:
-        if csources:
-            subprocess.check_call(args.pp + args.cflags + csources, stdout=out_file)
-        if cxxsources:
-            subprocess.check_call(args.pp + args.cxxflags + cxxsources, stdout=out_file)
+
+    def pp(flags):
+        def run(files):
+            return subprocess.check_output(args.pp + flags + files)
+
+        return run
+
+    cpus = multiprocessing.cpu_count()
```

Comment:
Should this check for the not implemented error? and make sure we have at least 1 cpu.

```
cpus = 1
try:
    cpus = multiprocessing.cpu_count()
except NotImplementedError:
    pass

if cpus > 1:
    cpus -= 1
```

Reply:
True (although this is supported just about everywhere, including Windows, Linux, Mac, *BSD). multiprocessing.cpu_count() is a wrapper around os.cpu_count() anyway, so implemented as

`cpus = os.cpu_count() or 1`